packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -414,8 +414,14 @@ class IOSSimulator extends Device {
     return false;
   }
 
+  String _simulatorHomePath;
+  String get simulatorHomePath {
+     _simulatorHomePath ??= platform.environment['IOS_SIMULATOR_HOME'] ?? homeDirPath;
+    return _simulatorHomePath;
+  }
+
   String get logFilePath {
-    return fs.path.join(homeDirPath, 'Library', 'Logs', 'CoreSimulator', id, 'system.log');
+    return fs.path.join(simulatorHomePath, 'Library', 'Logs', 'CoreSimulator', id, 'system.log');
```

Comment:
Tangentially-related, but what assumptions do we have around this file? For iOS 11 and later, I believe we're now using `/usr/bin/log` to capture simulator logs, and don't rely on this file at all.

Reply:
[Here](https://github.com/flutter/flutter/blob/277001d1a8678b211958608db41b6a8649f08835/packages/flutter_tools/lib/src/ios/simulators.dart#L474), we check the iOS version...

[Here](https://github.com/flutter/flutter/blob/277001d1a8678b211958608db41b6a8649f08835/packages/flutter_tools/lib/src/ios/simulators.dart#L440), we check that the file exists...

And [here](https://github.com/flutter/flutter/blob/277001d1a8678b211958608db41b6a8649f08835/packages/flutter_tools/lib/src/ios/simulators.dart#L523), we'll create it if it doesn't exist, which seems bad...

I updated the last of those to perform the version check.